app-text/mandoc/mandoc-1.14.3.recipe

```diff
@@ -0,0 +1,92 @@
+SUMMARY="A suite of tools compiling mdoc from the OpenBSD project"
+DESCRIPTION="A suite of tools compiling mdoc, the roff macro language of choice \
+for BSD manual pages, and man, the predominant historical language for UNIX \
+manuals. It is small, ISO C, ISC-licensed, and quite fast. The main component \
+of the toolset is the mandoc utility program, based on the libmandoc validating \
+compiler, to format output for UTF-8 and ASCII UNIX terminals, HTML 5, \
+PostScript, and PDF."
+HOMEPAGE="http://mandoc.bsd.lv/"
+COPYRIGHT="1989-2017 mandoc Authors"
+# Note: update LICENSE on every new version
+LICENSE="mandoc-$portVersion"
+REVISION="1"
+SOURCE_URI="http://mandoc.bsd.lv/snapshots/mandoc-$portVersion.tar.gz"
+CHECKSUM_SHA256="0b0c8f67958c1569ead4b690680c337984b879dfd2ad4648d96924332fd99528"
+PATCHES="mandoc-$portVersion.patchset"
+
+ARCHITECTURES="!x86_gcc2 x86_64"
+SECONDARY_ARCHITECTURES="x86"
+
+CONFLICTS="
+	cmd:man
+"
```

Comment:
@alaviss, you rock! It's great to have you with us, both as a GCI participant and as a mentor, which you also do very well!

And, of course, it is also great to have contributions outside the GCI contest!

== new topic ==

I haven't tested mandoc yet and I don't know what exactly you mean by "Allow to change /etc".
Haiku has a symlink for etc:
```
/etc -> /boot/system/settings/etc
```
so you could even leave that unchanged. But if you wish, you can use `finddir B_COMMON_ETC_DIRECTORY` (`/boot/system/settings/etc`) or `finddir B_SYSTEM_SETTINGS_DIRECTORY` (`/boot/system/settings`), or even `find_directory()`.
This is entirely up to you.

== new topic ==

We generally put the CONFLICTS bloc right after the REQUIRES bloc. BTW, the indentation of the closing double quotes needs a tab:
```diff
 CONFLICTS="
 	cmd:man
-"
+	"
```

Reply:
Sorry, I wasn't really clear back then. What I did was moving the `man.conf` to `B_SYSTEM_SETTINGS_DIRECTORY`

Reply:
Hmm, Haiku already has a `man.conf` (and a `man.conf.default`) in there (/boot/system/settings/) so it might only be possible to put that file in this directory if it is compatible with the man package.
BTW, would it make sense to co-install man and mandoc ? Or is the latter a replacement for the former?

== new topic ==

BTW, for the implementation you might try using find_directory.

Regarding the DB cache you mentioned earlier, I guess B_SYSTEM_CACHE_DIRECTORY is where the files should be generated.

Reply:
The latter is a replacement for the former, given that our man port is ages old and upstream have abandoned it

I will definitely use `find_directory()`, however, the code path is complex and need further analysis before I could make changes to mandoc

Reply:
We can easily add a GCI task for that so that you can easily have +1 in your counter of completed tasks. (If you add a 👍 to this comment I'll understand.)

Reply:
FWIW, the task is already created (but is not yet published.)
GCI mentors, if you wish to co-mentor (or only have a look at) that task, look for "[PUBLISH ME PLEASE]" in TASKS (and uncheck "MY TASKS")